utility/environment_check.py

```diff
@@ -0,0 +1,127 @@
+"""
+Util for environment check befora and after test to compare and find stale
+leftovers
+"""
+import logging
+import pytest
+
+from ocs import ocp, defaults, constants, exceptions
+from deepdiff import DeepDiff
+
+log = logging.getLogger(__name__)
+
+
+POD = ocp.OCP(kind=constants.POD)
+SC = ocp.OCP(kind=constants.STORAGECLASS)
+CEPHFILESYSTEM = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+CEPHBLOCKPOOL = ocp.OCP(kind=constants.CEPHBLOCKPOOL)
+PV = ocp.OCP(kind=constants.PV)
+PVC = ocp.OCP(kind=constants.PVC)
+SECRET = ocp.OCP(kind=constants.SECRET)
+NS = ocp.OCP(kind=constants.NAMESPACE)
+
+ADDED_RESOURCE = 'iterable_item_added'
+REMOVED_RESOURCE = 'iterable_item_removed'
+
+
+@pytest.fixture(scope='class')
+def environment_checker(request):
+    request.addfinalizer(get_status_after_execution)
+    get_status_before_execution()
+
+
+def get_status_before_execution():
+    """
+
+    """
+    defaults.ENV_STATUS_PRE['pod'] = POD.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['sc'] = SC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['cephfs'] = CEPHFILESYSTEM.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['cephbp'] = CEPHBLOCKPOOL.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['pv'] = PV.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['pvc'] = PVC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['secret'] = SECRET.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['namespace'] = NS.get(all_namespaces=True)['items']
+
+
+def get_status_after_execution():
+    """
+
+    """
+    defaults.ENV_STATUS_POST['pod'] = POD.get(all_namespaces=True)['items']
```

Comment:
Do you want to capture complete details about a resource ? How about simply printing (name, namespace, kind)

Reply:
yes,

for this initial PR, I'm just checking if new/old resources left/removed after the test finishes but in the future (for now, it is nice to have) we will add more functionality to check also changes in existing resources.
For example, if I'm modifying some existing resource, I want to ensure it is returning to its original values.

Reply:
sure